src/java.base/share/classes/java/nio/file/Files.java

```diff
@@ -2803,37 +2803,33 @@ public static Path walkFileTree(Path start,
             do {
                 FileVisitResult result;
                 switch (ev.type()) {
-                    case ENTRY :
+                    case ENTRY -> {
                         IOException ioe = ev.ioeException();
                         if (ioe == null) {
                             assert ev.attributes() != null;
                             result = visitor.visitFile(ev.file(), ev.attributes());
                         } else {
                             result = visitor.visitFileFailed(ev.file(), ioe);
                         }
-                        break;
-
-                    case START_DIRECTORY :
+                    }
+                    case START_DIRECTORY -> {
                         result = visitor.preVisitDirectory(ev.file(), ev.attributes());
 
                         // if SKIP_SIBLINGS and SKIP_SUBTREE is returned then
                         // there shouldn't be any more events for the current
                         // directory.
                         if (result == FileVisitResult.SKIP_SUBTREE ||
-                            result == FileVisitResult.SKIP_SIBLINGS)
+                                result == FileVisitResult.SKIP_SIBLINGS)
                             walker.pop();
-                        break;
-
-                    case END_DIRECTORY :
+                    }
+                    case END_DIRECTORY -> {
                         result = visitor.postVisitDirectory(ev.file(), ev.ioeException());
 
                         // SKIP_SIBLINGS is a no-op for postVisitDirectory
                         if (result == FileVisitResult.SKIP_SIBLINGS)
                             result = FileVisitResult.CONTINUE;
-                        break;
-
-                    default :
-                        throw new AssertionError("Should not get here");
+                    }
+                    default -> throw new AssertionError("Should not get here");
```

Comment:
This is subjective, and we're still finding our way with how best to construct some of the more complex switch expressions.

Where possible, I think it is best to remove assignments from the individual case branches. The use of `yield` makes it very clear what is going on, and ensure that each case branch, well... yields something. So how about:

```
  FileVisitResult result = switch (ev.type()) {
      case ENTRY -> {
          IOException ioe = ev.ioeException();
          if (ioe == null) {
              assert ev.attributes() != null;
              yield visitor.visitFile(ev.file(), ev.attributes());
          } else {
              yield visitor.visitFileFailed(ev.file(), ioe);
          }
      }
      case START_DIRECTORY -> {
          var r = visitor.preVisitDirectory(ev.file(), ev.attributes());

          // if SKIP_SIBLINGS and SKIP_SUBTREE is returned then
          // there shouldn't be any more events for the current
          // directory.
          if (r == FileVisitResult.SKIP_SUBTREE ||
              r == FileVisitResult.SKIP_SIBLINGS)
              walker.pop();
          yield r;
      }
      case END_DIRECTORY -> {
          var r = visitor.postVisitDirectory(ev.file(), ev.ioeException());
          // SKIP_SIBLINGS is a no-op for postVisitDirectory
          if (r == FileVisitResult.SKIP_SIBLINGS)
              r = FileVisitResult.CONTINUE;
          yield r;
      }
      default -> throw new AssertionError("Should not get here");
  };
```

Reply:
OK. I've made that change now. See 433b02a